osu.Game/Input/ConfineMouseTracker.cs

```diff
@@ -28,26 +28,37 @@ private void load(OsuGame game, FrameworkConfigManager frameworkConfigManager, O
         {
             frameworkConfineMode = frameworkConfigManager.GetBindable<ConfineMouseMode>(FrameworkSetting.ConfineMouseMode);
             frameworkWindowMode = frameworkConfigManager.GetBindable<WindowMode>(FrameworkSetting.WindowMode);
-            frameworkWindowMode.BindValueChanged(_ => updateConfineMode());
+            frameworkWindowMode.BindValueChanged(_ => updateGameConfineMode());
 
             osuConfineMode = osuConfigManager.GetBindable<OsuConfineMouseMode>(OsuSetting.ConfineMouseMode);
             localUserPlaying = game.LocalUserPlaying.GetBoundCopy();
 
-            osuConfineMode.ValueChanged += _ => updateConfineMode();
-            localUserPlaying.BindValueChanged(_ => updateConfineMode(), true);
+            osuConfineMode.ValueChanged += _ => updateFrameworkConfineMode();
+            localUserPlaying.BindValueChanged(_ => updateFrameworkConfineMode(), true);
         }
 
-        private void updateConfineMode()
+        private LeasedBindable<OsuConfineMouseMode> leasedOsuConfineMode;
+
+        private void updateGameConfineMode()
         {
-            // confine mode is unavailable on some platforms
-            if (frameworkConfineMode.Disabled)
-                return;
+            if (frameworkWindowMode.Value == WindowMode.Fullscreen && leasedOsuConfineMode == null)
+            {
+                leasedOsuConfineMode = osuConfineMode.BeginLease(true);
```

Comment:
not sure the lease is necessary - is there a specific reason for this being done? just comparing to other places in settings we do this kind of disabling and they don't seem to be leasing, not sure it's achieving much.

Reply:
It's mostly for visual reasons. This way, when entering fullscreen the dropdown will gray out and its text will say "Always" until the window mode is changed to windowed/borderless again, at which point the old value will be restored. Just felt nice, mostly.

Reply:
Oh, didn't realise that was the intention. I'm not sure then, since the setting is "confine mouse cursor to *window*".. is there even a window when running fullscreen?

Reply:
Hiding is also a viable way, sure. I think I slightly prefer disabling since it gives the user indication as to what's happening, but I'm not going to defend it much.

Reply:
I've reverted the leasing logic and made the dropdown hidden in full-screen, but I did keep it in the input section since I think it's still more apt there, considering the other options in that section (i.e. absolute input mapping toggle, among others).